### PR TITLE
[Translator] Adding more information about default translation domain 'messages'

### LINKED ISSUE #14390: The 'default' domain for translation files
I have spent a lot of time wondering why my new translation files are not being recognized.
The issue was that they were namend "en.yaml" "en.php" etc. (no domain defined)

The manual does not state that domain part is required. 

> domain: An **optional** way to organize messages into groups. Unless parts of the application are explicitly separated from each other, it is **recommended** to only use default messages domain;

Please give some information about that here.. https://symfony.com/doc/current/translation.html#translation-resource-locations

### PR DESCRIPTION
**Pull request by Matts**
Fixes #14390

Not 100% sure about the formulation, but just an idea how to make it more clear

### CODE REVIEW

**Comment by Matts** [comment]
Fixes #14390

File: translation.rst [hunk 1]
```diff
@@ -655,9 +655,12 @@ priority message files.
 The filename of the translation files is also important: each message file
 must be named according to the following path: ``domain.locale.loader``:
 
-* **domain**: An optional way to organize messages into groups. Unless
+* **domain**: Domains are a way to organize messages into groups. Unless
   parts of the application are explicitly separated from each other, it is
-  recommended to only use default ``messages`` domain;
+  recommended to only use default ``messages`` domain.
+
+If no domains are explicitly defined while using the translator, Symfony
```
Comment (noniagriconomie) [inline translation.rst]: i think you should use the same alignment no?

File: translation.rst [hunk 2]
```diff
@@ -655,9 +655,12 @@ priority message files.
...
 must be named according to the following path: ``domain.locale.loader``:
 
-* **domain**: An optional way to organize messages into groups. Unless
+* **domain**: Domains are a way to organize messages into groups. Unless
   parts of the application are explicitly separated from each other, it is
-  recommended to only use default ``messages`` domain;
+  recommended to only use default ``messages`` domain.
+
+If no domains are explicitly defined while using the translator, Symfony 
+will look for the ``messages`` domain (e.g. ``messages.en.yaml``)
```
Comment (noniagriconomie) [inline translation.rst]:
```suggestion
will default to the ``messages`` domain (e.g. ``messages.en.yaml``)
```

what about this?

**Comment by Matts** [comment]
Fixed @noniagriconomie

**Review by maxhelias — APPROVED** [review]

**Comment by wouterj** [comment]
Hi @Matts! Thanks for this PR. I've reduced the changes a bit after merging in ded8825 (the main problem points were "optional" and missing the file name, if I understand the issue correctly). Please let me know if you disagree :)

**Pull request merged**